enterprise/server/remote_execution/containers/firecracker/firecracker.go

```diff
@@ -500,20 +500,20 @@ func NewContainer(ctx context.Context, env environment.Env, task *repb.Execution
 		if *snaputil.EnableLocalSnapshotSharing {
 			runnerID = ""
 		}
-		c.snapshotKey, err = snaploader.NewKey(task, cd.GetHash(), runnerID)
+		c.snapshotKeys, err = snaploader.SnapshotKeys(task, cd.GetHash(), runnerID)
 		if err != nil {
 			return nil, err
 		}
 		// If recycling is enabled and a snapshot exists, then when calling
 		// Create(), load the snapshot instead of creating a new VM.
 
-		recyclingEnabled := platform.IsTrue(platform.FindValue(task.GetCommand().GetPlatform(), platform.RecycleRunnerPropertyName))
+		recyclingEnabled := platform.IsTrue(platform.FindValue(task.GetExecutionTask().GetCommand().GetPlatform(), platform.RecycleRunnerPropertyName))
 		if recyclingEnabled && *snaputil.EnableLocalSnapshotSharing {
-			_, err := loader.GetSnapshot(ctx, c.snapshotKey)
+			_, err := loader.GetSnapshot(ctx, c.snapshotKeys)
 			c.createFromSnapshot = (err == nil)
 		}
 	} else {
-		c.snapshotKey = opts.SavedState.GetSnapshotKey()
+		c.snapshotKeys = &fcpb.SnapshotKeys{MainKey: opts.SavedState.GetSnapshotKey()}
```

Comment:
not sure I have a better name but I found this confusing that snapshotKey**s** is a single proto rather than a slice of keys.

snapshotKeySet or SnapshotIdentifier or something? idk

Reply:
Yeah this did feel mildly awkward. I forgot to come back around and rename it. SnapshotKeySet sgtm - done